packages/cli/lib/services/model.service.ts

```diff
@@ -139,6 +139,20 @@ class ModelService {
             if (hasUndefined) {
                 tsType = `${tsType} | undefined`;
             }
+
+            if (tsType.includes('|')) {
+                const types = tsType.split('|');
+                const isEnum = types.every((type) => !JAVASCRIPT_PRIMITIVES.includes(type.trim()));
+
+                if (isEnum) {
+                    const enumValues = tsType
+                        .split('|')
+                        .map((e) => `'${e.trim()}'`)
+                        .join(' | ');
+                    tsType = enumValues;
+                }
+            }
```

Comment:
I think this logic could be generalized even if tsType doesn't include `|`. After splitting we would operate on a single element list. It also has the benefit of handling stuff like boolean vs bool, ...
```
return tsTypes
   .split('|')
   .map((e) => {
      let res = e;
      switch (e.trim()) {
         case 'boolean':
         case 'bool':
            res = 'boolean';
       ....
      }
      return `'${res}'`
   })
   .join('|')
```

Reply:
We're not explicitly handling union types other than the enum (string literal union type) here and I'd like to stay away from that in the context of this PR.